images/Makefile

```diff
@@ -1,37 +1,38 @@
 # WHAT specifies the OS image to build
 WHAT?=
+IS_LATEST?=
 DOCKER_USER?=weaveworks
 VERSION?=$(shell git describe HEAD --tags)
+RELEASE?=latest
+TAG:=${RELEASE}$(if $(strip $(VERSION)),-${VERSION})
+OP:=build
 
 all: build
 build:
 ifeq ($(WHAT),)
 	$(error WHAT is a required argument)
 endif
 	@ls ${WHAT} >/dev/null
-	docker build -t ${DOCKER_USER}/ignite-${WHAT}:${VERSION} ${WHAT}
+	docker build --build-arg RELEASE -t ${DOCKER_USER}/ignite-${WHAT}:${TAG} ${WHAT}
 
 push:
-	docker push ${DOCKER_USER}/ignite-${WHAT}:${VERSION}
-	docker tag ${DOCKER_USER}/ignite-${WHAT}:${VERSION} ${DOCKER_USER}/ignite-${WHAT}:latest
+	docker push ${DOCKER_USER}/ignite-${WHAT}:${TAG}
+	docker tag ${DOCKER_USER}/ignite-${WHAT}:${TAG} ${DOCKER_USER}/ignite-${WHAT}:${RELEASE}
+	docker push ${DOCKER_USER}/ignite-${WHAT}:${RELEASE}
+ifeq ($(IS_LATEST),true)
+	docker tag ${DOCKER_USER}/ignite-${WHAT}:${TAG} ${DOCKER_USER}/ignite-${WHAT}:latest
 	docker push ${DOCKER_USER}/ignite-${WHAT}:latest
-
-build-all:
-	# Temporarily comment out the kernel build for the RC
-	#	make -C kernel
-	$(MAKE) build WHAT=amazon-kernel
-	$(MAKE) build WHAT=amazonlinux
-	$(MAKE) build WHAT=alpine
-	$(MAKE) build WHAT=ubuntu
-	$(MAKE) build WHAT=centos
-	$(MAKE) build WHAT=kubeadm
+endif
 
 push-all: build-all
-	# Temporarily comment out the kernel build for the RC
-	#	make -C kernel push
-	$(MAKE) push WHAT=amazon-kernel
-	$(MAKE) push WHAT=amazonlinux
-	$(MAKE) push WHAT=alpine
-	$(MAKE) push WHAT=ubuntu
-	$(MAKE) push WHAT=centos
-	$(MAKE) push WHAT=kubeadm
+	$(MAKE) OP=push build-all
+
+build-all:
+	$(MAKE) ${OP} WHAT=amazon-kernel
+	$(MAKE) ${OP} WHAT=amazonlinux		RELEASE=2		IS_LATEST=true
+	$(MAKE) ${OP} WHAT=alpine
+	$(MAKE) ${OP} WHAT=ubuntu			RELEASE=16.04	IS_LATEST=true
+	$(MAKE) ${OP} WHAT=ubuntu			RELEASE=18.04	IS_LATEST=true
+	$(MAKE) ${OP} WHAT=ubuntu			RELEASE=19.04	IS_LATEST=true
```

Comment:
there can only be one latest tag. remove IS_LATEST from 19.04 and 16.04

Reply:
Whoops, copy-pasted them a bit too quickly :smile:. Done in 
91e375e.